decoder/candidates.go

```diff
@@ -177,6 +177,13 @@ func nameTokenRangeAtPos(tokens hclsyntax.Tokens, pos hcl.Pos) (hcl.Range, error
 			}
 			return hcl.Range{}, fmt.Errorf("token is %s, not Ident", t.Type.String())
 		}
+		// EOF has zero length
```

Comment:
What does this mean, EOF has zero length?

Reply:
I can tune the comment, it means the length of the _EOF token_ is zero, because it's not really anything visible in the editor.

This causes `ContainsPos()` to always return `false` since it's looking in a zero-length range.

Reply:
How can an EOF _token_ have zero length? Wouldn't the existence of a token, have at least length 1? 🤔

Reply:
This is not byte length, it's the length of the HCL range, so basically a _(visible) character_ slice length. EOF is technically invisible, similar to BOM and other special bytes. You wouldn't see those rendered in a typical editor.